settings/src/main/java/org/odk/collect/settings/ODKAppSettingsImporter.kt

```diff
@@ -28,6 +29,10 @@ class ODKAppSettingsImporter(
     )
 
     fun fromJSON(json: String, project: Project.Saved): Boolean {
-        return settingsImporter.fromJSON(json, project)
+        return try {
+            settingsImporter.fromJSON(json, project)
+        } catch (e: JSONException) {
+            false
```

Comment:
I'm not sure how this could actually happen - currently it's not covered by a test. I could definitely be missing a case, but as far as I can tell, the changes you've made to `JsonSchemaSettingsValidator` (rejecting invalid JSON) should prevent a `JSONException` from occurring later?

Reply:
I'm not able to fully remember my thinking since it's been a while but in my comment above I also said that:
> I can't come up with any json that would not be caught in JsonSchemaSettingsValidator but cause JSONException

so probably you are right and we can get rid of it at least for now.